spec/lib/forecast_factory_spec.rb

```diff
@@ -65,5 +65,100 @@
         expect(forecast.temperature.max).to eq(16.6)
       end
     end
+
+    context "when a _zone_ ID is given" do
+      context "and the zone ID is present in the forecast_representation" do
+        let(:forecast_representation_from_api) do
+          {"forecastdate" => "02-10-2024 15:38",
+           "timestamp" => 1727883524517.199,
+           "zones" => [
+             {"forecasts" => [
+                {"NO2" => 1,
+                 "O3" => 2,
+                 "PM10" => 1,
+                 "PM2.5" => 1,
+                 "forecast_date" => "2024-10-02",
+                 "non_pollution_version" => nil,
+                 "pollen" => -999,
+                 "pollution_version" => 202410021550,
+                 "rain_am" => 0.81,
+                 "rain_pm" => 0.66,
+                 "temp_max" => 16.6,
+                 "temp_min" => 10.0,
+                 "total" => 10,
+                 "total_status" => "VERY HIGH",
+                 "uv" => 2,
+                 "wind_am" => 4.9,
+                 "wind_pm" => 5.3}
+              ],
+              "zone_id" => 29,
+              "zone_name" => "Southwark",
+              "zone_type" => 1},
+
+             {"forecasts" => [
+                {"NO2" => 3,
+                 "O3" => 3,
+                 "PM10" => 3,
+                 "PM2.5" => 3,
+                 "forecast_date" => "2024-10-02",
+                 "non_pollution_version" => nil,
+                 "pollen" => -999,
+                 "pollution_version" => 202410021550,
+                 "rain_am" => 0.33,
+                 "rain_pm" => 0.3,
+                 "temp_max" => 33.3,
+                 "temp_min" => 13.3,
+                 "total" => 3,
+                 "total_status" => "LOW",
+                 "uv" => 3,
+                 "wind_am" => 33.3,
+                 "wind_pm" => 3.33}
+              ],
+              "zone_id" => 2,
+              "zone_name" => "Barnet",
+              "zone_type" => 1}
+           ]}
+        end
+
+        it "builds forecasts for that zone" do
+          forecast = ForecastFactory.build(forecast_representation_from_api, 2).first
+
+          aggregate_failures do
```

Comment:
`aggregate_failures {}` will run all of the assertions rather than stop after the first failure. I find that helpful in most cases to get a report of all the problems.

Normally we like the idea of "1 assertion per test" see http://blog.jayfields.com/2007/06/testing-one-assertion-per-test.html as this makes for clear specifications. However, it sometimes makes sense to document several elements of an individual point in the spec. Where there isn't real value in defining separate test examples....

Reply:
ah, thanks